storage/txcache/txListBySenderMap.go

```diff
@@ -73,8 +75,10 @@ func (txMap *txListBySenderMap) addSender(sender string) *txListForSender {
 }
 
 // This function should only be called in a critical section managed by a "txListForSender"
-func (txMap *txListBySenderMap) notifyScoreChange(txList *txListForSender) {
-	txMap.backingMap.NotifyScoreChange(txList)
+func (txMap *txListBySenderMap) notifyScoreChange(txList *txListForSender, scoreParams senderScoreParams) {
+	score := txMap.scoreComputer.computeScore(scoreParams)
+	txList.setLastComputedScore(score)
+	txMap.backingMap.NotifyScoreChange(txList, score)
 }
```

Comment:
line 106 removeSender can act incorrectly if called concurrently for same sender, e.g removing the only tx for that sender and removing the sender due to low tx both will call removeSender. The removal is done only once but the counter is decremented twice.

Reply:
Good point! As above, added a TODO. Is this ok?